RecoLocalCalo/HGCalRecProducers/plugins/HGCalRecHitMapProducer.cc

```diff
@@ -0,0 +1,71 @@
+#define __RecoLocalCalo_HGCRecProducers_HGCalRecHitMapProducer_H__
+
+// user include files
+#include <map>
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/global/EDProducer.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+
+#include "DataFormats/HGCRecHit/interface/HGCRecHitCollections.h"
+
+class HGCalRecHitMapProducer : public edm::global::EDProducer<> {
+public:
+  HGCalRecHitMapProducer(const edm::ParameterSet&);
+  ~HGCalRecHitMapProducer() override {}
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+  void produce(edm::StreamID, edm::Event&, const edm::EventSetup&) const override;
+
+private:
+  edm::EDGetTokenT<HGCRecHitCollection> hits_ee_token;
+  edm::EDGetTokenT<HGCRecHitCollection> hits_fh_token;
+  edm::EDGetTokenT<HGCRecHitCollection> hits_bh_token;
+};
+
+DEFINE_FWK_MODULE(HGCalRecHitMapProducer);
+
+HGCalRecHitMapProducer::HGCalRecHitMapProducer(const edm::ParameterSet& ps) {
+  hits_ee_token = consumes<HGCRecHitCollection>(ps.getParameter<edm::InputTag>("HGCEEInput"));
+  hits_fh_token = consumes<HGCRecHitCollection>(ps.getParameter<edm::InputTag>("HGCFHInput"));
+  hits_bh_token = consumes<HGCRecHitCollection>(ps.getParameter<edm::InputTag>("HGCBHInput"));
+  produces<std::map<DetId, const HGCRecHit*>>();
+}
+
+void HGCalRecHitMapProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+
+  desc.add<edm::InputTag>("HGCEEInput", edm::InputTag("HGCalRecHit", "HGCEERecHits"));
+  desc.add<edm::InputTag>("HGCFHInput", edm::InputTag("HGCalRecHit", "HGCHEFRecHits"));
+  desc.add<edm::InputTag>("HGCBHInput", edm::InputTag("HGCalRecHit", "HGCHEBRecHits"));
```

Comment:
is "HGC" prefix needed? can this be confused with something else?
If not ->
```suggestion
  desc.add<edm::InputTag>("EEInput", {"HGCalRecHit", "HGCEERecHits"});
  desc.add<edm::InputTag>("FHInput", {"HGCalRecHit", "HGCHEFRecHits"});
  desc.add<edm::InputTag>("BHInput", {"HGCalRecHit", "HGCHEBRecHits"});
```